test/multipart-json.test.js

```diff
@@ -0,0 +1,334 @@
+    } catch (error) {
+      t.error(error, 'formData request pump: no err')
+    }
+  })
+})
+
+test('should always reject JSON parsing if the value was truncated', function (t) {
+  t.plan(2)
+
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+
+  fastify.register(multipart, { limits: { fieldSize: 2 } })
+
+  fastify.post('/', async function (req, reply) {
+    try {
+      for await (const part of req.parts()) {
+        t.type(part.value, 'string')
+      }
+
+      reply.code(200).send()
+    } catch (error) {
+      t.ok(error instanceof fastify.multipartErrors.InvalidJSONFieldError)
+      reply.code(500).send()
+    }
+  })
+
+  fastify.listen(0, async function () {
+    // request
+    const form = new FormData()
+    const opts = {
+      protocol: 'http:',
+      hostname: 'localhost',
+      port: fastify.server.address().port,
+      path: '/',
+      headers: form.getHeaders(),
+      method: 'POST'
+    }
+
+    const req = http.request(opts, res => {
+      t.equal(res.statusCode, 500)
+      res.resume()
+      res.on('end', () => {
+        t.pass('res ended successfully')
+      })
+    })
+
+    form.append('object', JSON.stringify({ a: 'b' }), { contentType: 'application/json' })
+
+    try {
+      await pump(form, req)
+    } catch (error) {
+      t.error(error, 'formData request pump: no err')
+    }
+  })
+})
+
+test('should be able to use JSON schema to validate request when value is a string', function (t) {
+  t.plan(5)
+
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+
+  fastify.register(multipart, { attachFieldsToBody: true, sharedSchemaId: '#mySharedSchema' })
+
+  fastify.post(
+    '/',
+    {
+      schema: {
+        body: {
+          type: 'object',
+          required: ['field'],
+          properties: {
+            field: {
+              allOf: [{ $ref: '#mySharedSchema' }, { properties: { value: { type: 'string' } } }]
+            }
+          }
+        }
+      }
+    },
+    async function (req, reply) {
+      t.ok(req.isMultipart())
+
+      t.same(Object.keys(req.body), ['field'])
+      t.equal(req.body.field.value, '{"a":"b"}')
+
+      reply.code(200).send()
+    }
+  )
+
+  fastify.listen(0, async function () {
+    // request
+    const form = new FormData()
+    const opts = {
+      protocol: 'http:',
+      hostname: 'localhost',
+      port: fastify.server.address().port,
+      path: '/',
+      headers: form.getHeaders(),
+      method: 'POST'
+    }
+
+    const req = http.request(opts, res => {
+      t.equal(res.statusCode, 200)
+      res.resume()
+      res.on('end', () => {
+        t.pass('res ended successfully')
+      })
+    })
+
+    form.append('field', JSON.stringify({ a: 'b' }))
+
+    try {
+      await pump(form, req)
+    } catch (error) {
+      t.error(error, 'formData request pump: no err')
+    }
+  })
+})
+
+test('should be able to use JSON schema to validate request when value is a JSON', function (t) {
```

Comment:
Could we test the `400` status code case?

Reply:
If I got what you meant, it should be there now.